caas/kubernetes/provider/k8s.go

```diff
@@ -227,7 +227,7 @@ func (k *kubernetesClient) ensureNamespaceAnnotationForControllerUUID(controller
 	if err != nil && !errors.IsNotFound(err) {
 		return errors.Trace(err)
 	}
-	if ns != nil && !k8sannotations.New(ns.Annotations).HasAll(k.annotations) {
+	if ns != nil && !isLegacy && !k8sannotations.New(ns.Annotations).HasAll(k.annotations) {
```

Comment:
Instead of removing this check, can we default to checking, but only not if isLegacy is set?

Long term, we need a way to get off of legacy, and be able to add the new annotations (even if we keep the old annotations for compatibility, at some point the model needs to move to new things so we don't have to support legacy forever).

But for *this* PR, I think we could just do:

```
if !legacy {
  ns, err := 
  ...
  if ns != nil && !k8sannotations.New()

}
```

(I don't think we need to load the namespace if we aren't going to check it.

Reply:
ns is used below too, so loading it always is fine. I think just a simple restructure like @jameinel recommended would be good to simplify that condition.